regex-assembly/932270.ra

```diff
@@ -0,0 +1,8 @@
+##! Please refer to the documentation at
+##! https://coreruleset.org/docs/development/regex_assembly/.
+
+##! Bash - Tilde Expansion
+
+~\+
```

Comment:
```suggestion
~\+\d*
```

Reply:
Why a (*) and not a (+)? This was earlier removed because it is already covered by `~\+`.

Reply:
Because both of these are valid:
`~+`
`~+5`

Reply:
Yeah, but `~\+` will also catch `~\+5`

Reply:
But you're right, since it's an empty match, it's already covered. However, you should add a condition for what can come after, since `~+s`, for example, doesn't work. a "non-word" token is required. This works for example: `~+${}`, because the expansion is the empty string. So you need something like `~\+[\d\s]`

Reply:
Okay, but why remove the existing regex, it is also a valid match, while adding the `\d` won't match the original tilde expansion payload of PWD and OLDPWD. I think we should add both as was my initial commit.

Reply:
@dune73 any suggestions on how we should handle this? I still think we should go by what you initially said.

Reply:
You misunderstand. They expression would match both because of the end of string anchor (although the expression I wrote isn't correct). But to make it clearer, you could use these two, which is equivalent:
```
~\+$
~\+[\d\s]+
```